features/zones/Create.feature

```diff
@@ -148,3 +151,68 @@ Feature: Zones: Create Zone flow
 
     Then the "$error" element contains "Invalid zone name"
     And the "$instructions" element doesn't exist
+
+  Scenario: Exiting the form in a safe state without confirm dialog
+    Given the environment
+      """
+      KUMA_SUBSCRIPTION_COUNT: 1
+      """
+    And the URL "/provision-zone" responds with
+      """
+      body:
+        token: spat_595QOxTSreRmrtdh8ValuoeUAzXMfBmRwYU3V35NQvwgLAWIU
+      """
+    And the URL "/zones+insights/test" responds with
+      """
+      body:
+        zoneInsight:
+          subscriptions:
+            - connectTime: '2020-07-28T16:18:09.743141Z'
+              disconnectTime: ~
+              status: {}
+      """
+
+    When I visit the "/zones/create" URL
+    And I "type" "test" into the "$name-input" element
+    And I click the "$create-zone-button" element
+
+    Then the "$instructions" element exists
+    And the "$zone-connected-scanner" element contains "The Zone “test” is now connected"
+
+    When I click the "$exit-button" element
+
+    Then the "$confirm-exit-modal" element doesn't exist
+    And the page title contains "Zone Control Planes"
+
+  Scenario: Exiting the form in an unsafe state with confirm dialog
+    Given the environment
+      """
+      KUMA_SUBSCRIPTION_COUNT: 0
+      """
+    And the URL "/provision-zone" responds with
+      """
+      body:
+        token: spat_595QOxTSreRmrtdh8ValuoeUAzXMfBmRwYU3V35NQvwgLAWIU
+      """
+    And the URL "/zones+insights/test" responds with
+      """
+      body:
+        zoneInsight:
+          subscriptions: []
+      """
+
+    When I visit the "/zones/create" URL
+    And I "type" "test" into the "$name-input" element
+    And I click the "$create-zone-button" element
+
+    Then the "$instructions" element exists
+    Then the "$zone-connected-scanner" element contains "Waiting for Zone to be connected"
```

Comment:
I'm kinda assuming this will never change due to subscriptions being zero length, is that right?

Reply:
No, the scanner can also be in the waiting state.

I’m using these assertions as a kind of insurance for the test having enough time to actually see this UI before trying to click Exit. When working on this, I occasionally had scenarios where these tests failed because the test clicked Exit too early (i.e. in the short gap between clicking the create button and the conditions for changing the exit button interaction). If the scanner shows the waiting state, these conditions are up-to-date, too.

Reply:
Not sure I follow but I think from what you said the assumption was right, subscriptions=0 seems to mean that the zone will never be connected and therefore when the text clicks the exit button we are guaranteed to trigger the warning. Feel free to resolve.

Reply:
That’s not the case: the scanner isn’t immediately in the complete/success state but first in the loading state. At that point, the condition for triggering the confirm modal is _also_ not in effect yet and we would exit directly without the confirmation.

Reply:
I think we are talking about different things here. I am talking about after this message is visible. You are talking about before the message it visible.

Reply:
Okay so once it switched to the waiting state, it will remain in that state on account of the zero subscriptions.